apps/android/app/src/main/java/chat/simplex/app/views/chat/ChatView.kt

```diff
@@ -118,11 +101,22 @@ fun ChatView(chatModel: ChatModel) {
           }
         }
       },
-      openDirectChat = { contactId ->
+      showChat = { apiId, chatType, pagination ->
         val c = chatModel.chats.firstOrNull {
-          it.chatInfo is ChatInfo.Direct && it.chatInfo.contact.contactId == contactId
+          it.chatInfo.chatType == chatType && it.chatInfo.apiId == apiId
+        }
+        if (c != null) {
+          // If there are more unread messages than pagination.count wants, load all unread
+          if (pagination is ChatPagination.Last && c.chatStats.unreadCount > pagination.count) {
```

Comment:
this block of code actually shows why it is not approach - first you have to pass this end then to use if to decide what to do - it just needs two different functions.